File: cinema-backend/src/main/java/org/example/cinemabackend/cinema/application/dto/response/ScreeningRoomResponse.java
package org.example.cinemabackend.cinema.application.dto.response;

import lombok.Builder;

import java.util.Set;

@Builder
public record ScreeningRoomResponse(
        Long id,
        String name,
        SeatResponse[][] seats,
        Set<ProjectionTechnologyResponse> supportedTechnologies
) {
}

File: cinema-customer-frontend/src/app/repertory/dtos/response/screening-room.response.ts
import {ProjectionTechnologyResponse} from "./projection-technology.response";
import {SeatResponse} from "./seat.response";

export class ScreeningRoomResponse {
  constructor(
    readonly name: string,
    readonly seats: SeatResponse[][],
    readonly supportedTechnologies: ProjectionTechnologyResponse[],
  ) {
  }
}
